src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1445,35 +1445,62 @@ private enum GetBytesMode
                     break;
             }
 
-            int curByte = isBigEndian ? length - 1 : 0;
+            int curByte = isBigEndian ? length : 0;
             int increment = isBigEndian ? -1 : 1;
 
             if (bits != null)
             {
-                for (int i = 0; i < bits.Length - 1; i++)
+                if (BitConverter.IsLittleEndian && sign > 0)
                 {
-                    uint dword = bits[i];
+                    ReadOnlySpan<byte> srcBytes = MemoryMarshal.AsBytes(bits.AsSpan(..^1));
 
-                    if (sign == -1)
+                    if (isBigEndian)
+                    {
+                        curByte = length - srcBytes.Length;
+                        Span<byte> destBytes = destination.Slice(curByte, srcBytes.Length);
+                        srcBytes.CopyTo(destBytes);
+                        destBytes.Reverse();
```

Comment:
This doesn't look correct. Under big endian, the bytes should be reserved for every 4 byte (every uint32), not as a whole. See also `BigIntegerHexParser.TryParseWholeBlocks`.